src/binxml/tokens.rs

```diff
@@ -169,45 +127,57 @@ pub fn read_template<'a>(
                     )
                 })?;
         }
-        substitution_array.push(value);
+        substitution_array.push(BinXMLDeserializedTokens::Value(value));
     }
 
     Ok(BinXmlTemplate {
-        definition: template_def,
+        template_def_offset: template_definition_data_offset,
         substitution_array,
     })
 }
 
+pub fn read_template_data_size(cursor: &mut Cursor<&[u8]>) -> Result<u32> {
```

Comment:
For sake of readability - we should export this to a struct
```
BinXmlTemplateDefinitionLink {
    // offset to next node
    next_offset: Offset,
    guid: Guid,
    data_size: u32
}
```

and then refer to this as `read_template_link`.

The tuple `BinXmlTemplateDefinition` + `BinXmlTemplateDefinitionLink` is what is logically a "TemplateNode".
This would make the code iterating over the linked list clearer.

I could add a docstring in `Chunk` explaining this layout.